CoreScriptsRoot/Modules/Panel3D.lua

```diff
@@ -9,36 +13,124 @@ local Panel3D = {}
 
 Panel3D.Panels = {
 	Lower = 1,
-	Hamburger = 2
+	Hamburger = 2,
+	Settings = 3
+}
+
+Panel3D.Orientation = {
+	Horizontal = 1,
+	Fixed = 2
+}
+
+Panel3D.Visibility = {
+	BelowAngleThreshold = 1,
+	Modal = 2
 }
 
 local panelDefaultVectors = {
 	[Panel3D.Panels.Lower] = CFrame.Angles(math.rad(-45), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5)),
-	[Panel3D.Panels.Hamburger] = CFrame.Angles(math.rad(-55), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5))
-}
-local panelLockThreshold = {
-	[Panel3D.Panels.Lower] = math.rad(-25),
-	[Panel3D.Panels.Hamburger] = math.rad(-25)
+	[Panel3D.Panels.Hamburger] = CFrame.Angles(math.rad(-55), 0, 0):vectorToWorldSpace(Vector3.new(0, 0, -5)),
+	[Panel3D.Panels.Settings] = Vector3.new(0, 0, -SETTINGS_DISTANCE)
 }
+local panelLockThreshold = math.rad(-25)
 local panelTransparencyBias = { --tuned values; raise the opacity value to this power
 	[Panel3D.Panels.Lower] = 6.5,
-	[Panel3D.Panels.Hamburger] = 8 
+	[Panel3D.Panels.Hamburger] = 8,
+	[Panel3D.Panels.Settings] = 0
 }
 local panels = {}
 
 local renderStepName = "Panel3D"
 
-
 local cursor = Instance.new("ImageLabel")
 cursor.Image = "rbxasset://textures/Cursors/Gamepad/Pointer.png"
 cursor.Size = UDim2.new(0, 8, 0, 8)
 cursor.BackgroundTransparency = 1
 cursor.ZIndex = 10
 
-local menuOpened = false
+local resetOrientationFlag = false
+local modalPanelIsOpen = false
 
+local GuiService = game:GetService("GuiService")
 local UserInputService = game:GetService("UserInputService")
 
+local cursorHidden = false
+local lastMouseMove = tick()
+
+local function autoHideCursor(hide)
+	if not UserInputService.VREnabled then
+		cursorHidden = false
+		UserInputService.MouseIconTransparency = 0
+		return
+	end
+	if hide then
+		--don't hide if there's a tool in the character
+		local character = game.Players.LocalPlayer.Character
+		if character then
+			for i, v in ipairs(character:GetChildren()) do
+				if v:IsA("Tool") then
+					autoHideCursor(false)
+					lastMouseMove = tick()
+					return
+				end
+			end
+		end
+		cursorHidden = true
+		spawn(function()
+			pcall(function() 
+				while UserInputService.MouseIconTransparency < 1 and cursorHidden do
+					local dt = game:GetService("RunService").RenderStepped:wait()
+					UserInputService.MouseIconTransparency = math.min(1, UserInputService.MouseIconTransparency + dt / CURSOR_FADE_TIME)
+				end
+			end)
+			UserInputService.OverrideMouseIconBehavior = Enum.OverrideMouseIconBehavior.ForceHide
+		end)
+	else
+		cursorHidden = false
+		pcall(function()
+			UserInputService.MouseIconTransparency = 0
```

Comment:
It interferes with developer options. If your script hides the mouse icon and we really want to make sure it's shown (the settings menu), your code would have to deal with it being changed on you.